### PR TITLE
Handle response from HMI during resumption data

### PR DESCRIPTION
**Pull request by ValeriiMalkov**
Add RPC control button on settings view
Add view for ever app of error handling

### CODE REVIEW

File: ffw/BasicCommunicationRPC.js [hunk 1]
```diff
@@ -77,11 +77,79 @@ FFW.BasicCommunication = FFW.RPCObserver
       onAppUnregisteredNotification: 'BasicCommunication.OnAppUnregistered',
       onSDLCloseNotification: 'BasicCommunication.OnSDLClose',
       onResumeAudioSourceNotification: 'BasicCommunication.OnResumeAudioSource',
+
+      appContainer:Em.Object.create({}),
+
+      upDateRpc: function(appID){
```
Comment (AKalinich-Luxoft) [inline ffw/BasicCommunicationRPC.js]: @ValeriiMalkov update is one word, so rename to `updateRpc`

File: app/model/sdl/Abstract/data.js [hunk 2]
```diff
@@ -152,7 +161,7 @@ SDL.SDLModelData = Em.Object.create(
      *
      * @type {Object}
      */
-    policyURLs: [],
+    policyURLs: [],
```
Comment (AKalinich-Luxoft) [inline app/model/sdl/Abstract/data.js]: @ValeriiMalkov revert this change

File: app/model/sdl/Abstract/AppModel.js [hunk 3]
```diff
@@ -325,32 +325,45 @@ SDL.ABSAppModel = Em.Object.extend(
...
-        }
-        console.log(commands.length);
-        if (request.id >= 0) {
-          FFW.UI.sendUIResult(
-            SDL.SDLModel.data.resultCode.SUCCESS, request.id,
-            request.method
+      result = 
+              SDL.SDLModel.data.resultCode[FFW.BasicCommunication.appContainer[this.appName]['uiAddCommand']];
+      
+      if(result == SDL.SDLModel.data.resultCode.SUCCESS ||
```
Comment (AKalinich-Luxoft) [inline app/model/sdl/Abstract/AppModel.js]: @ValeriiMalkov suggest to create function in RPCObserver.js:
```
isSuccessResultCode: function(resultCode) {
  return [SDL.SDLModel.data.resultCode.SUCCESS, SDL.SDLModel.data.resultCode.WARNINGS].includes(resultCode);
}
```

And just call this function from here and other places like this

File: ffw/BasicCommunicationRPC.js [hunk 4]
```diff
@@ -77,11 +77,79 @@ FFW.BasicCommunication = FFW.RPCObserver
       onAppUnregisteredNotification: 'BasicCommunication.OnAppUnregistered',
       onSDLCloseNotification: 'BasicCommunication.OnSDLClose',
       onResumeAudioSourceNotification: 'BasicCommunication.OnResumeAudioSource',
+
+      appContainer:Em.Object.create({}),
```
Comment (AKalinich-Luxoft) [inline ffw/BasicCommunicationRPC.js]: @ValeriiMalkov why is this container here? Please move it to the base class (RPCObserver.js)

File: ffw/BasicCommunicationRPC.js [hunk 5]
```diff
@@ -77,11 +77,79 @@ FFW.BasicCommunication = FFW.RPCObserver
...
+
+      upDateRpc: function(appID){
+       for(key in this.appContainer[appID]){
+       this.set('rpcStruct.' + key,this.appContainer[appID][key]);
+       }
+      },
+
+      saveButton: function(){
```
Comment (AKalinich-Luxoft) [inline ffw/BasicCommunicationRPC.js]: @ValeriiMalkov please move all these functions to RPCObserver as well

File: app/model/sdl/Abstract/AppModel.js [hunk 6]
```diff
@@ -325,32 +325,45 @@ SDL.ABSAppModel = Em.Object.extend(
...
-          SDL.SDLController.buttonsSort(parentID, this.appID);
-          SDL.OptionsView.commands.refreshItems();
-        }
-        console.log(commands.length);
-        if (request.id >= 0) {
-          FFW.UI.sendUIResult(
-            SDL.SDLModel.data.resultCode.SUCCESS, request.id,
-            request.method
+      result = 
+              SDL.SDLModel.data.resultCode[FFW.BasicCommunication.appContainer[this.appName]['uiAddCommand']];
```
Comment (AKalinich-Luxoft) [inline app/model/sdl/Abstract/AppModel.js]: @ValeriiMalkov quite complicated. Please create a separate function in RPCObserver:
```
getCustomResultCode: function(appID, method) {
  code = this.appContainer[appID][method];
  return SDL.SDLModel.data.resultCode[code];
}
```
and just use this function from places like this one:
```
result = FFW.RPCObserver.getCustomResultCode(this.appId, 'uiAddCommand');
```

File: app/model/sdl/Abstract/AppModel.js [hunk 7]
```diff
@@ -325,32 +325,45 @@ SDL.ABSAppModel = Em.Object.extend(
...
+            'Adding more than 1000 item to the top menu or to submenu is not allowed.'
           );
         }
       } else {
         FFW.UI.sendError(
-          SDL.SDLModel.data.resultCode.REJECTED, request.id,
+          result, request.id,
           request.method,
-          'Adding more than 1000 item to the top menu or to submenu is not allowed.'
+          'Customer chose erroneous response.'
```
Comment (AKalinich-Luxoft) [inline app/model/sdl/Abstract/AppModel.js]: @ValeriiMalkov please reword as above

File: app/model/sdl/Abstract/AppModel.js [hunk 8]
```diff
@@ -393,32 +406,44 @@ SDL.ABSAppModel = Em.Object.extend(
       // parentID is equal to 'top' cause Top level menu ID
       var parentID = 'top';
       var commands = this.get('commandsList.' + parentID);
+
+      result = 
+              SDL.SDLModel.data.resultCode[FFW.BasicCommunication.appContainer[this.appName]['AddSubmenu']];
```
Comment (AKalinich-Luxoft) [inline app/model/sdl/Abstract/AppModel.js]: @ValeriiMalkov please use a separate function like above

File: app/model/sdl/Abstract/AppModel.js [hunk 9]
```diff
@@ -393,32 +406,44 @@ SDL.ABSAppModel = Em.Object.extend(
...
+      if(result == SDL.SDLModel.data.resultCode.SUCCESS || 
+                result == SDL.SDLModel.data.resultCode.WARNINGS) {
```
Comment (AKalinich-Luxoft) [inline app/model/sdl/Abstract/AppModel.js]: @ValeriiMalkov please use a separate function like above

File: app/model/sdl/Abstract/AppModel.js [hunk 10]
```diff
@@ -393,32 +406,44 @@ SDL.ABSAppModel = Em.Object.extend(
...
-          SDL.SDLModel.data.resultCode.SUCCESS, request.id,
-          request.method
-        );
       } else {
         FFW.UI.sendError(
-          SDL.SDLModel.data.resultCode.REJECTED, request.id,
+          result, request.id,
           request.method,
-          'Adding more than 1000 item to the top menu or to submenu is not allowed.'
+          'Customer chose erroneous response.'
```
Comment (AKalinich-Luxoft) [inline app/model/sdl/Abstract/AppModel.js]: @ValeriiMalkov please reword like above

File: app/model/sdl/Abstract/data.js [hunk 11]
```diff
@@ -33,6 +33,15 @@
 
 SDL.SDLModelData = Em.Object.create(
   {
+    
+    init: function(){
+      this.resultCodes = Object.keys(this.resultCode);
+    },
+
+    /**
+    *  List of result code as array
```
Comment (AKalinich-Luxoft) [inline app/model/sdl/Abstract/data.js]: @ValeriiMalkov Array of result codes

File: app/model/sdl/Abstract/data.js [hunk 12]
```diff
@@ -449,6 +458,18 @@ SDL.SDLModelData = Em.Object.create(
       'GENERIC_ERROR': 22,
       'USER_DISALLOWED': 23
     },
+
+    vehicleDataResultCode: [
```
Comment (AKalinich-Luxoft) [inline app/model/sdl/Abstract/data.js]: @ValeriiMalkov please add description what is this

File: app/view/settings/RPCControlView.js [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        elementId: 'rpc_settings',
+        classNames: 'in_settings_view',
+        classNameBindings: [
+            'SDL.States.settings.rpccontrol.active:active_state:inactive_state'
+        ],
+        childViews: [
+            'AppsList'
+          ],
+          showAppList: function() {
```
Comment (AKalinich-Luxoft) [inline app/view/settings/RPCControlView.js]: @ValeriiMalkov please fix identation here

File: app/view/settings/RPCControlView.js [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+            elementId: 'rpc_settings_list',
+           classNames: 'rpc_settings_list',
+            itemsOnPage: 5,
+            items: new Array()
+          }
+        )
+
+
+    }
+);
```
Comment (AKalinich-Luxoft) [inline app/view/settings/RPCControlView.js]: @ValeriiMalkov please add a blank line in the end

File: app/view/settings/RPCControls/RPCControlConfigView.js [hunk 15]
```diff
@@ -0,0 +1,744 @@
+/*
+ * Copyright (c) 2013, Ford Motor Company All rights reserved.
```
Comment (AKalinich-Luxoft) [inline app/view/settings/RPCControls/RPCControlConfigView.js]: @ValeriiMalkov 2018

File: app/view/settings/RPCControls/RPCControlConfigView.js [hunk 16]
```diff
@@ -0,0 +1,744 @@
...
+        childViews: [
+            'backButton',
+            'rpcList',
+            'resultCodeTracker',
+            'saveButton',
+            'resetButton',
+            'appNameLabel'
+          ],
+         
+          appNameLabel: SDL.Label.extend(
```
Comment (AKalinich-Luxoft) [inline app/view/settings/RPCControls/RPCControlConfigView.js]: @ValeriiMalkov please fix identation

File: app/view/settings/RPCControls/RPCControlConfigView.js [hunk 17]
```diff
@@ -0,0 +1,744 @@
...
+          backButton: SDL.Button.extend(
+            {
+              classNames: [
+                'backControl'
+              ],
+              action: 'onState',
+              target: 'SDL.SettingsController',
+              goToState: 'rpccontrol',
+              icon: 'images/media/ico_back.png',
+              style: 'top: 100px',
```
Comment (AKalinich-Luxoft) [inline app/view/settings/RPCControls/RPCControlConfigView.js]: @ValeriiMalkov please do not hardcode a style with the view. Use css files for style formatting

File: app/view/settings/RPCControls/RPCControlConfigView.js [hunk 18]
```diff
@@ -0,0 +1,744 @@
...
+              'createInteractionChoiceSetSelect',
+              'subscribeWayPointsSelect',
+              'vehicleDataGlobalSelect'
+              ],
+              vehicleDataGlobalSelect: Em.Select.extend(
+                {
+                  elementId: 'vehicleDataGlobalSelect',
+                  classNames: 'vehicleDataGlobalSelect',
+                  contentBinding: 'SDL.SDLModel.data.resultCodes',
+                  //valueBinding: 'FFW.BasicCommunication.rpcStruct.vrAddCommand'
```
Comment (AKalinich-Luxoft) [inline app/view/settings/RPCControls/RPCControlConfigView.js]: @ValeriiMalkov should it be removed?

File: app/view/settings/RPCControls/RPCControlConfigView.js [hunk 19]
```diff
@@ -0,0 +1,744 @@
...
+                  elementId: 'subscribeWayPointsSelect',
+                  classNames: 'subscribeWayPointsSelect',
+                  contentBinding: 'SDL.SDLModel.data.resultCodes',
+                  valueBinding: 'FFW.BasicCommunication.rpcStruct.SubscribeWayPoints'
+                }
+              ),
+            }
+          )
+    }
+);
```
Comment (AKalinich-Luxoft) [inline app/view/settings/RPCControls/RPCControlConfigView.js]: @ValeriiMalkov please add an empty line in the end of file

File: app/view/settingsView.js [hunk 20]
```diff
@@ -83,7 +85,8 @@ SDL.SettingsView = Em.ContainerView.create(
               'policies',
               'HMISettings',
               'light',
-              'seat'
+              'seat',
+	            'rpcControl'
```
Comment (AKalinich-Luxoft) [inline app/view/settingsView.js]: @ValeriiMalkov fix identation

File: css/settings.css [hunk 21]
```diff
@@ -8,8 +8,413 @@
...
+ }
+ .backControl {
+    top: 20px;
+    width: 35px;
+    height: 27px;
+    border: 1px solid #333;
+    border-radius: 2px;
+    left: 10px;
+}
+ .saveButton{
```
Comment (AKalinich-Luxoft) [inline css/settings.css]: @ValeriiMalkov try to separate styles with a blank lines between them

File: css/settings.css [hunk 22]
```diff
@@ -8,8 +8,413 @@
...
+   }
+   .subscribeVehicleDataSelections .driverBraking{
+
+     direction: initial;
+    left: 30px;
+    position: absolute;
+    top:610px;
+   }
+   .subscribeVehicleDataSelections .wiperStatus{
+
```
Comment (AKalinich-Luxoft) [inline css/settings.css]: @ValeriiMalkov please remove redundant empty line inside the syles

File: ffw/VRRPC.js [hunk 23]
```diff
@@ -113,15 +113,34 @@ FFW.VR = FFW.RPCObserver.create(
     /*
      * handle RPC requests here
      */
+    checkRequestType: function(type){
+      switch(type){
+        case 'Command':
+          return 'vrAddCommand';
+        case 'Choice':
+          return 'createInteractionChoiceSet';
+        default: break;
```
Comment (AKalinich-Luxoft) [inline ffw/VRRPC.js]: @ValeriiMalkov return an empty string for default case

File: ffw/VRRPC.js [hunk 24]
```diff
@@ -113,15 +113,34 @@ FFW.VR = FFW.RPCObserver.create(
...
       Em.Logger.log('FFW.VR.onRPCRequest');
       if (this.validationCheck(request)) {
         switch (request.method) {
           case 'VR.AddCommand':
           {
-            SDL.SDLModel.addCommandVR(request.params);
+            app = SDL.SDLController.getApplicationModel(request.params.appID);
+            var key = this.checkRequestType(request.params.type);
+            result = 
+              SDL.SDLModel.data.resultCode[FFW.BasicCommunication.appContainer[app.appName][key]];
```
Comment (AKalinich-Luxoft) [inline ffw/VRRPC.js]: @ValeriiMalkov please use a common function as above

File: ffw/VRRPC.js [hunk 25]
```diff
@@ -113,15 +113,34 @@ FFW.VR = FFW.RPCObserver.create(
...
+
+            if(result == SDL.SDLModel.data.resultCode.SUCCESS || 
+                result == SDL.SDLModel.data.resultCode.WARNINGS){
```
Comment (AKalinich-Luxoft) [inline ffw/VRRPC.js]: @ValeriiMalkov please usa a common function above

File: ffw/VRRPC.js [hunk 26]
```diff
@@ -359,7 +378,8 @@ FFW.VR = FFW.RPCObserver.create(
         return;
       }
       Em.Logger.log('FFW.' + method + 'Response');
-      if (resultCode === SDL.SDLModel.data.resultCode.SUCCESS) {
+      if (resultCode === SDL.SDLModel.data.resultCode.SUCCESS || 
+        resultCode === SDL.SDLModel.data.resultCode.WARNINGS) {
```
Comment (AKalinich-Luxoft) [inline ffw/VRRPC.js]: @ValeriiMalkov please use a common function as above

File: ffw/VRRPC.js [hunk 27]
```diff
@@ -371,6 +391,10 @@ FFW.VR = FFW.RPCObserver.create(
           }
         };
         this.client.send(JSONMessage);
+      } else {
+        this.sendError(
+          resultCode, id, method,
+          '');
```
Comment (AKalinich-Luxoft) [inline ffw/VRRPC.js]: @ValeriiMalkov why empty info parameter? Why not in a single line?

File: index.html [hunk 28]
```diff
@@ -227,6 +227,12 @@
 <script type="text/javascript" src="app/view/settings/policiesView.js"></script>
 <script type="text/javascript" src="app/view/settings/HMISettingsView.js"></script>
 
+<!-- RPCControls view-->
+<script type="text/javascript"
+        src="app/view/settings/RPCControls/RPCControlConfigView.js"></script>
+<script type="text/javascript" src="app/view/settings/RPCControlView.js"></script>
+
```
Comment (AKalinich-Luxoft) [inline index.html]: @ValeriiMalkov redundant empty line

File: app/controller/NavigationController.js [hunk 29]
```diff
@@ -128,19 +128,31 @@ SDL.NavigationController = Em.Object.create(
...
+          );
+        }
       } else {
         FFW.Navigation.sendError(
-          SDL.SDLModel.data.resultCode.REJECTED,
+          result,
           request.id,
           request.method,
-          'SDL Should not send this request more than once'
+          'Customer chose erroneous response'
```
[the remark this answers is not in the document]
Comment (ValeriiMalkov) [reply app/controller/NavigationController.js]: @AKalinich-Luxoft Fixed 5edc8064384166c13ea1414c2fc337b7cf7d99b8

File: ffw/BasicCommunicationRPC.js (hunk 1 above)
Comment (ValeriiMalkov) [reply ffw/BasicCommunicationRPC.js]: @AKalinich-Luxoft Fixed 5edc8064384166c13ea1414c2fc337b7cf7d99b8

File: app/model/sdl/Abstract/AppModel.js (hunk 3 above)
Comment (ValeriiMalkov) [reply app/model/sdl/Abstract/AppModel.js]: @AKalinich-Luxoft Fixed e6d3971ca37d9f4d52f761a59a31f2e5408fa3cc

File: ffw/VRRPC.js (hunk 24 above)
Comment (ValeriiMalkov) [reply ffw/VRRPC.js]: @AKalinich-Luxoft Fixed e6d3971ca37d9f4d52f761a59a31f2e5408fa3cc

File: ffw/VRRPC.js (hunk 25 above)
Comment (ValeriiMalkov) [reply ffw/VRRPC.js]: @AKalinich-Luxoft Fixed e6d3971ca37d9f4d52f761a59a31f2e5408fa3cc

File: ffw/VRRPC.js (hunk 26 above)
Comment (ValeriiMalkov) [reply ffw/VRRPC.js]: @AKalinich-Luxoft Fixed e6d3971ca37d9f4d52f761a59a31f2e5408fa3cc

File: app/controller/NavigationController.js [hunk 30]
```diff
@@ -128,19 +128,31 @@ SDL.NavigationController = Em.Object.create(
...
-        this.model.set('isSubscribedOnWayPoints', true);
-        FFW.Navigation.sendNavigationResult(
-          SDL.SDLModel.data.resultCode.SUCCESS,
-          request.id,
-          request.method
-        );
+      app = SDL.SDLController.getApplicationModel(request.params.appID);
+      result = 
+              SDL.SDLModel.data.resultCode[FFW.BasicCommunication.appContainer[app.appName]['SubscribeWayPoints']];
+      if(result == SDL.SDLModel.data.resultCode.SUCCESS || result == SDL.SDLModel.data.resultCode.WARNINGS){
```
[the remark this answers is not in the document]
Comment (ValeriiMalkov) [reply app/controller/NavigationController.js]: @AKalinich-Luxoft Fixed e6d3971ca37d9f4d52f761a59a31f2e5408fa3cc

File: app/controller/NavigationController.js [hunk 31]
```diff
@@ -128,19 +128,31 @@ SDL.NavigationController = Em.Object.create(
...
+        if (!this.model.isSubscribedOnWayPoints) {
```
[the remark this answers is not in the document]
Comment (ValeriiMalkov) [reply app/controller/NavigationController.js]: @mked-luxoft it was replaced into function in commit e6d3971ca37d9f4d52f761a59a31f2e5408fa3cc

File: app/controller/SettingsController.js [hunk 32]
```diff
@@ -47,6 +47,10 @@ SDL.SettingsController = Em.Object.create(
     currentDeviceAllowance: null,
     onState: function(event) {
       SDL.States.goToStates('settings.' + event.goToState);
+      if('rpccontrol.rpcconfig' == event.goToState){
+        SDL.RPCControlConfigView.set('appNameLabel.content',event.appName);
```
[the remark this answers is not in the document]
Comment (ValeriiMalkov) [reply app/controller/SettingsController.js]: @mked-luxoft it was replaced into function in commit e6d3971ca37d9f4d52f761a59a31f2e5408fa3cc

File: app/model/sdl/Abstract/AppModel.js [hunk 33]
```diff
@@ -325,32 +325,45 @@ SDL.ABSAppModel = Em.Object.extend(
...
-        if (request.id >= 0) {
-          FFW.UI.sendUIResult(
-            SDL.SDLModel.data.resultCode.SUCCESS, request.id,
-            request.method
+      result = 
+              SDL.SDLModel.data.resultCode[FFW.BasicCommunication.appContainer[request.params.appName]['uiAddCommand']];
+      
+      if(result == SDL.SDLModel.data.resultCode.SUCCESS || 
+        result == SDL.SDLModel.data.resultCode.WARNINGS){
+
```
[the remark this answers is not in the document]
Comment (ValeriiMalkov) [reply app/model/sdl/Abstract/AppModel.js]: @mked-luxoft it was replaced into function in commit e6d3971ca37d9f4d52f761a59a31f2e5408fa3cc

File: app/model/sdl/Abstract/AppModel.js [hunk 34]
```diff
@@ -393,32 +406,44 @@ SDL.ABSAppModel = Em.Object.extend(
       // parentID is equal to 'top' cause Top level menu ID
       var parentID = 'top';
       var commands = this.get('commandsList.' + parentID);
+
+      result = 
+              SDL.SDLModel.data.resultCode[FFW.BasicCommunication.appContainer[request.params.appName]['AddSubmenu']];
+      if(result == SDL.SDLModel.data.resultCode.SUCCESS || 
+                result == SDL.SDLModel.data.resultCode.WARNINGS) {
       // Magic number is limit of 1000 commands added on one menu
```
[the remark this answers is not in the document]
Comment (ValeriiMalkov) [reply app/model/sdl/Abstract/AppModel.js]: @mked-luxoft it was replaced into function in commit e6d3971ca37d9f4d52f761a59a31f2e5408fa3cc

File: ffw/VRRPC.js [hunk 35]
```diff
@@ -113,15 +113,34 @@ FFW.VR = FFW.RPCObserver.create(
...
           {
-            SDL.SDLModel.addCommandVR(request.params);
+            app = SDL.SDLController.getApplicationModel(request.params.appID);
+            var key = this.checkRequestType(request.params.type);
+            result = 
+              SDL.SDLModel.data.resultCode[FFW.BasicCommunication.appContainer[app.appName][key]];
+
+            if(result == SDL.SDLModel.data.resultCode.SUCCESS || 
+                result == SDL.SDLModel.data.resultCode.WARNINGS){
+              SDL.SDLModel.addCommandVR(request.params);
```
[the remark this answers is not in the document]
Comment (ValeriiMalkov) [reply ffw/VRRPC.js]: @mked-luxoft it was replaced into function in commit e6d3971ca37d9f4d52f761a59a31f2e5408fa3cc

File: css/settings.css [hunk 36]
```diff
@@ -8,8 +8,466 @@
...
+   }
+
+   .subscribeVehicleDataSelections .fuelRange{
+    direction: initial;
+    left: 30px;
+    position: absolute;
+    top:935px;
+   }
+
+   .subscribeVehicleDataSelections .gps{
```
Comment (AKalinich-Luxoft) [inline css/settings.css]: @ValeriiMalkov please fix indentation for this style and above

File: ffw/BasicCommunicationRPC.js [hunk 37]
```diff
@@ -77,11 +77,124 @@ FFW.BasicCommunication = FFW.RPCObserver
...
+        };
+        for(key in this.currentVehicleDataStruct) {
+          this.set('currentVehicleDataStruct.' + key, 'SUCCESS');
+       };
+      },
+
+      /*
+      * setCurrentAppID function. Sets of currentAppID parameter
+      */
+      setCurrentAppID: function(appID){
```
Comment (AKalinich-Luxoft) [inline ffw/BasicCommunicationRPC.js]: @ValeriiMalkov update/save/reset functions also could be moved to RPCObserver

File: app/controller/NavigationController.js [hunk 38]
```diff
@@ -128,19 +128,30 @@ SDL.NavigationController = Em.Object.create(
...
      */
     subscribeWayPoints: function(request) {
-      if (!this.model.isSubscribedOnWayPoints) {
-        this.model.set('isSubscribedOnWayPoints', true);
-        FFW.Navigation.sendNavigationResult(
-          SDL.SDLModel.data.resultCode.SUCCESS,
-          request.id,
-          request.method
-        );
+      result =
```
Comment (AKalinich-Luxoft) [inline app/controller/NavigationController.js]: @ValeriiMalkov quite complicated. Please create a separate function in RPCObserver:
```
getCustomResultCode: function(appID, method) {
  code = this.appContainer[appID][method];
  return SDL.SDLModel.data.resultCode[code];
}
```
and just use this function from places like this one:
```
result = FFW.RPCObserver.getCustomResultCode(this.appId, 'uiAddCommand');
```

File: ffw/VRRPC.js [hunk 39]
```diff
@@ -113,15 +113,33 @@ FFW.VR = FFW.RPCObserver.create(
...
         switch (request.method) {
           case 'VR.AddCommand':
           {
-            SDL.SDLModel.addCommandVR(request.params);
+            app = SDL.SDLController.getApplicationModel(request.params.appID);
+            var key = this.checkRequestType(request.params.type);
+            result = 
+              SDL.SDLModel.data.resultCode[FFW.BasicCommunication.appContainer[app.appID][key]];
+
+            if(FFW.BasicCommunication.isSuccessResultCode(result)){
```
Comment (AKalinich-Luxoft) [inline ffw/VRRPC.js]: @ValeriiMalkov why are you adressing to BasicCommunication? Just use `FFW.RPCObserver.isSuccessResultCode(result)`

File: css/settings.css (hunk 36 above)
Comment (ValeriiMalkov) [reply css/settings.css]: @AKalinich-Luxoft Fixed in d5ebbf359313dfcfdcab221c2ea7ea9857ffbd12

File: ffw/BasicCommunicationRPC.js (hunk 37 above)
Comment (ValeriiMalkov) [reply ffw/BasicCommunicationRPC.js]: @AKalinich-Luxoft Fixed in d5ebbf359313dfcfdcab221c2ea7ea9857ffbd12

File: app/controller/NavigationController.js (hunk 38 above)
Comment (ValeriiMalkov) [reply app/controller/NavigationController.js]: @AKalinich-Luxoft Fixed in d5ebbf359313dfcfdcab221c2ea7ea9857ffbd12

File: ffw/VRRPC.js (hunk 39 above)
Comment (ValeriiMalkov) [reply ffw/VRRPC.js]: @AKalinich-Luxoft Fixed in d5ebbf359313dfcfdcab221c2ea7ea9857ffbd12

File: ffw/RPCObserver.js [hunk 40]
```diff
@@ -29,6 +29,131 @@
...
+     /*
+     * saveButton function. Saves parameters after pressing save button
+     * on the RPCControl view and returns to app list
+     */
+     saveButton: function(){
+       var appName = SDL.RPCControlConfigView.appNameLabel.content;
+       app = SDL.SDLModel.data.registeredApps.filterProperty(
+         'appName',
+         appName
+       )[0];
```
Comment (ZhdanovP) [inline ffw/RPCObserver.js]: @ValeriiMalkov In my vision, will be better to put array in some var like `filterProp` and than take a first element `app = filterProp[0]`. Moreover what will be if filterProperty return empty array?

File: ffw/RPCObserver.js [hunk 41]
```diff
@@ -29,6 +29,131 @@
...
+       for(key in this.appContainer[app.appID]){
+         this.set('appContainer.'+ app.appID + '.'+key, this.rpcStruct[key]);
```
Comment (ZhdanovP) [inline ffw/RPCObserver.js]: @ValeriiMalkov Need separate spaces between the operators `+`

File: ffw/TTSRPC.js [hunk 42]
```diff
@@ -377,6 +382,9 @@ FFW.TTS = FFW.RPCObserver.create(
           }
         };
         this.client.send(JSONMessage);
+      } else {
+        this.sendError(
+          resultCode, id, method, '');
```
Comment (ZhdanovP) [inline ffw/TTSRPC.js]: May be on one line?

File: ffw/UIRPC.js [hunk 43]
```diff
@@ -215,9 +215,13 @@ FFW.UI = FFW.RPCObserver.create(
...
+          resultCode = FFW.UI.getCustomResultCode(request.params.appID, 'uiSetGlobalProperties');
+          
+          if(FFW.UI.isSuccessResultCode(resultCode)){
             SDL.SDLModel.setProperties(request.params);
-            this.sendUIResult(
-              SDL.SDLModel.data.resultCode.SUCCESS, request.id, request.method
+          }
+          this.sendUIResult(
+            resultCode, request.id, request.method
             );
```
Comment (ZhdanovP) [inline ffw/UIRPC.js]: Sometimes you use `);` on the same line with arguments, sometimes on the separate. I think It can be on one line with `sendUIResult`
